From e2a5a966c2fae74469a5b33cb5937938af357471 Mon Sep 17 00:00:00 2001
From: John <reidjr@lineone.net>
Date: Wed, 21 May 2014 05:52:00 +1000
Subject: [PATCH 3/3] Don't unecessarily query the backend if we don't need it

No point attempting to skip a commercial if there are none to skip.
This prevents querying the backend for an up to date frames count

Ref #12016

Signed-off-by: Jean-Yves Avenard <jyavenard@mythtv.org>
---
 mythtv/libs/libmythtv/mythplayer.cpp |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/mythplayer.cpp b/mythtv/libs/libmythtv/mythplayer.cpp
index 150235d..0fc596b 100644
--- a/mythtv/libs/libmythtv/mythplayer.cpp
+++ b/mythtv/libs/libmythtv/mythplayer.cpp
@@ -3089,7 +3089,8 @@ void MythPlayer::EventLoop(void)
     // Handle automatic commercial skipping
     uint64_t jumpto = 0;
     if (deleteMap.IsEmpty() && (ffrew_skip == 1) &&
-       (kCommSkipOff != commBreakMap.GetAutoCommercialSkip()))
+       (kCommSkipOff != commBreakMap.GetAutoCommercialSkip()) &&
+        commBreakMap.HasMap())
     {
         QString msg;
         uint64_t frameCount = GetCurrentFrameCount();
-- 
1.7.10.2

